client/src/components/ui/callout.tsx

```diff
@@ -1,7 +1,9 @@
+import React from "react";
+
 import { EditIcon, InfoIcon } from "./icons";
 
 interface CalloutProps {
-  text: string;
+  text: React.ReactNode;
```

Comment:
If you want links in the callout, set `onClick`, this will probably break where ever `Callout` is used

Reply:
no it shouldn't be nullable, was it modified somewhere?

Reply:
I notice that if it’s empty string the error-callout component still takes position, but if the text is null it won’t.Since user input is correct, the error-callout should be hided. So I tried to set error messages as <string | null>

Reply:
If the error callout should be hidden then you shouldn't be rendering it at all. Don't hide it by messing with the string parameter, hide it by not rendering it. Other pages kept a state `errorOccured` and if it's set to `true` then the error callout is rendered